sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreReader.java

```diff
@@ -216,16 +220,23 @@ private Flux<List<StoreResult>> readFromReplicas(List<StoreResult> resultCollect
             return Flux.error(new GoneException());
         }
         List<Pair<Flux<StoreResponse>, Uri>> readStoreTasks = new ArrayList<>();
+        Map<Uri, String> replicaStatusesAttempting = new HashMap<>();
+        Map<Uri, String> replicaStatusesNotAttempting = new HashMap<>();
 
+        resolveApiResults.forEach(uri -> {
+            replicaStatusesNotAttempting.put(uri, uri.getHealthStatusDiagnosticString());
```

Comment:
Why first always add then remove again - wouldn't you be able to first publish the attempting list and then only add to non-attempting for addresses not already in attempting?

Reply:
Changed to adding the replicas not already in attempting as suggested.